homeassistant/components/ista_ecotrend/sensor.py

```diff
@@ -0,0 +1,250 @@
+        key=IstaSensorEntity.WATER,
+        translation_key=IstaSensorEntity.WATER,
+        device_class=SensorDeviceClass.WATER,
+        native_unit_of_measurement=UnitOfVolume.CUBIC_METERS,
+        state_class=SensorStateClass.TOTAL,
+        suggested_display_precision=1,
+        entity_registry_enabled_default=False,
+        consumption_type="water",
+    ),
+    IstaSensorEntity.WATER_COST: IstaSensorEntityDescription(
+        key=IstaSensorEntity.WATER_COST,
+        translation_key=IstaSensorEntity.WATER_COST,
+        device_class=SensorDeviceClass.MONETARY,
+        native_unit_of_measurement="EUR",
+        state_class=SensorStateClass.TOTAL,
+        suggested_display_precision=0,
+        entity_registry_enabled_default=False,
+        consumption_type="water",
+        value_type="costs",
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: IstaConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the ista EcoTrend sensors."""
+
+    coordinator = config_entry.runtime_data
+
+    async_add_entities(
+        IstaSensor(coordinator, description, consumption_unit)
+        for description in SENSOR_DESCRIPTIONS.values()
+        for consumption_unit in coordinator.data
+    )
+
+
+class IstaSensor(CoordinatorEntity[IstaCoordinator], SensorEntity):
+    """Ista EcoTrend sensor."""
+
+    entity_description: IstaSensorEntityDescription
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: IstaCoordinator,
+        entity_description: IstaSensorEntityDescription,
+        consumption_unit: str,
+    ) -> None:
+        """Initialize the ista EcoTrend sensor."""
+        super().__init__(coordinator)
+        self.consumption_unit = consumption_unit
+        self.entity_description = entity_description
+        self._attr_unique_id = f"{consumption_unit}_{entity_description.key}"
+        self._attr_device_info = DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            manufacturer="ista SE",
+            model="ista EcoTrend",
+            name=f"{coordinator.details[consumption_unit]["address"]["street"]} "
+            f"{coordinator.details[consumption_unit]["address"]["houseNumber"]}".strip(),
+            configuration_url="https://ecotrend.ista.de/",
+            identifiers={(DOMAIN, consumption_unit)},
+        )
+
+    @property
+    def native_value(self) -> StateType:
+        """Return the state of the device."""
+
+        return get_native_value(
+            data=self.coordinator.data[self.consumption_unit],
+            consumption_type=self.entity_description.consumption_type,
+            value_type=self.entity_description.value_type,
+        )
+
+    def _handle_coordinator_update(self) -> None:
+        """Handle coordinator update."""
+        self.hass.async_create_task(self.update_statistics())
+
+    async def update_statistics(self) -> None:
```

Comment:
do you need that in from the beginning, or can that be a separate PR? It's just a lot of extra code, especially with the `utils`